internal/workflow/processing.go

```diff
@@ -410,7 +416,92 @@ func (w *ProcessingWorkflow) SessionHandler(
 		return err
 	}
 
-	// Bundle transfer as an Archivematica standard transfer.
+	// Classify the SIP.
+	{
+		activityOpts := withActivityOptsForLocalAction(sessCtx)
+		var result activities.ClassifyPackageActivityResult
+		err := temporalsdk_workflow.ExecuteActivity(
+			activityOpts,
+			activities.ClassifyPackageActivityName,
+			activities.ClassifyPackageActivityParams{Path: tinfo.TempPath},
+		).Get(activityOpts, &result)
+		if err != nil {
+			return fmt.Errorf("classify package activity: %v", err)
+		}
+
+		tinfo.PackageType = result.Type
+	}
+
+	// If the SIP is a BagIt Bag, validate it.
+	if tinfo.IsDir && tinfo.PackageType == enums.PackageTypeBagIt {
+		id, err := w.createPreservationTask(
+			sessCtx,
+			datatypes.PreservationTask{
+				Name:   "Validate Bag",
+				Status: enums.PreservationTaskStatusInProgress,
+				StartedAt: sql.NullTime{
+					Time:  temporalsdk_workflow.Now(sessCtx).UTC(),
+					Valid: true,
+				},
+				PreservationActionID: tinfo.PreservationActionID,
+			},
+		)
+		if err != nil {
+			return err
+		}
+
+		// Validate the bag.
+		activityOpts := withActivityOptsForLocalAction(sessCtx)
+		var result bagit_activity.ValidateActivityResult
+		err = temporalsdk_workflow.ExecuteActivity(
+			activityOpts,
+			bagit_activity.ValidateActivityName,
+			&bagit_activity.ValidateActivityParams{Path: tinfo.TempPath},
+		).Get(activityOpts, &result)
+		if err != nil || !result.Valid {
+			note := "System error"
```

Comment:
Yeah, I'm not very happy with how the error propagation worked out here, but it's difficult to change with all the layers (bagit_gython, temporal_activities, enduro).  I was just looking at doing a check like `error.Is(err, bagit_gython.ErrInvalid)` in the temporal_activities, but because the validator is injected it seems wrong to check for `bagit_gython.ErrInvalid` specifically. But then checking specifically for `bagit_gython.ErrInvalid` here seems like bad coupling of the validator implementation (bagit_gython) to this layer. :confused:

Reply:
I thought it would be possible to do `err != nil` then "System error" or `!result.Valid` then "Validation error", but I'm not fully aware ATM about the final outcome of that discussion about error types you and Sevein had. Feel free to leave it as it is if you think that's okay, we'll see if there is really a need to differentiate. Maybe for now just don't use "System error".

Reply:
But that's a validation error, that's what I mean, if we are not able to differentiate them, I'd call them "Validation error" instead of "System error".

Reply:
I need to update https://github.com/artefactual-sdps/temporal-activities/blob/main/bagit/validate_activity.go, but my plan for the returns is:

1. `bagit_activity.ValidateActivityResult{Valid: true}, nil` == validation successful, no errors!
2. `bagit_activity.ValidateActivityResult{Valid: false, Error: "invalid: ..."}, nil` == validation failed, but no system errors
3. `nil, errors.New("some error")` == system error

In case 2 `if result.Error != "" {` will evaluate true, and overwrite the "System error" note. 
In case 3 `if result.Error != "" {` will evaluate false, and the note will be "System error"